### DISCUSSION TITLE
Question on compiling on Linux

### DISCUSSION THREAD

**Discussion by GuilleNM** [OpenFAST/openfast#1118] [Q&A]
Hi, I am trying to install OpenFast ona Ubuntu virtual machine and I was following the intallation steps on the openfast.readthedocs guide and I keep getting this error which I cannot find a way to fix. When I try to compile eveything it starts to compile each item but it freezes at 85% and after a while it gives that error. I tried installing and updating all the other tools and libraries and they seem to be up to date. Any idea of how to fix this issue?

----

[ 79%] Built target icedynlib
[ 84%] Built target icefloelib
[ 85%] Built target maplib
[ 85%] Building Fortran object modules/openfast-library/CMakeFiles/openfast_prelib.dir/src/FAST_Types.f90.o
gfortran: fatal error: Killed signal terminated program f951
compilation terminated.
make[2]: *** [modules/openfast-library/CMakeFiles/openfast_prelib.dir/build.make:63: modules/openfast-library/CMakeFiles/openfast_prelib.dir/src/FAST_Types.f90.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1766: modules/openfast-library/CMakeFiles/openfast_prelib.dir/all] Error 2
make: *** [Makefile:130: all] Error 2

----

Thank you

**Comment by andrew-platt** [comment]
Hi @GuilleNM

Which version of gfortran are you using?  And which branch/version of OpenFAST?  How much memory do you have available on the virtual machine?

I suspect that gfortran runs out of memory during compilation and fails.  The FAST_Types.f90 takes a lot of memory during compilation.  If this is the actual cause, there are two possible solutions:

1. Increase the amount of memory in the virtual machine
2. reduce the optimization on the compilation of the FAST_Types.f90 file.  Depending on exactly which branch or version of OpenFAST, this may already be reduced to `-O2` for the Intel compiler suite, but may also be needed for the gfortran compiler you are using on this VM.  This setting is in the `modules/openfast-library/CMakeLists.txt` file (lines 72:75). You can comment out the if statement so that the `set_source_file_properties` statement applies to all compilers, and see if this helps.

Hope this helps,
Andy

**Comment by rafmudaf** [comment]
@GuilleNM Do you need to compile the code? Check out this section of the installation documentation: https://openfast.readthedocs.io/en/main/source/install/index.html#download-binaries
